magda-typescript-common/src/JsonConnector.ts

```diff
@@ -265,33 +275,32 @@ export interface ConnectorSource {
      *
      * @returns {AsyncPage<any[]>} A page of datasets.
      */
-    getJsonDatasets(): AsyncPage<any[]>;
+    getJsonDatasets(): AsyncPage<DatasetContainer[]>;
 
     /**
      * Get a particular dataset given its ID.
-     *
      * @param {string} id The ID of the dataset.
-     * @returns {Promise<any>} The dataset object with the given ID.
+     * @returns {Promise<DatasetTags>} The dataset object with the given ID.
```

Comment:
Looks like the return type is `Promise<DatasetContainer>`, not `Promise<DatasetTags>`.